platform/measurement_utils.cpp

```diff
@@ -245,6 +245,40 @@ string FormatSpeedUnits(Units units)
   UNREACHABLE();
 }
 
+// Interleaves the bits of two 32-bit numbers.
+// The result is known as a Morton code.
+long interleave_bits(long x, long y)
+{
+  long c = 0;
+  for(int i=31; i>=0; i--)
+  {
+    c = (c << 1) | ((x >> i) & 1);
+    c = (c << 1) | ((y >> i) & 1);
+  }
+
+  return c;
+}
+
+string FormatOsmLink(double lat, double lon, int zoom)
+{
+  long x = round((lon + 180.0) / 360.0 * (1L<<32));
+  long y = round((lat + 90.0) / 180.0 * (1L<<32));
```

Comment:
Seems like better to use lround here.
https://en.cppreference.com/w/cpp/numeric/math/round

Reply:
I don't use `lround` because `x` variable should be `uint32_t`. BitwiseMerge function expects `uint32_t` type.